enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,242 @@
+reviewers:
+  - "@Miciah" ## reviewer for cluster-ingress-operator component
+  - "@tgeer" ## reviewer for CFE
+approvers:
+  - "@jerpeter1" ## approver for CFE
+  - "@Miciah" ## approver for cluster-ingress-operator component
+api-approvers:
+  - None
+creation-date: 2023-11-02
+last-updated: 2023-11-02
+tracking-link:
+  - https://issues.redhat.com/browse/RFE-2182
+  - https://issues.redhat.com/browse/OCPSTRAT-431
+  - https://issues.redhat.com/browse/NE-761
+see-also:
+  - None
+replaces:
+  - None
+superseded-by:
+  - None
+---
+
+# Support for admin configured CA trust bundle in Ingress Operator
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of OpenShift
+to configure trusted Certificate Authorities(CA) to verify the service's 
+certificate by OpenShift Router for the re-encrypted traffic.
+
+## Motivation
+
+Openshift administrator when creating a route for a service with re-encrypt termination
+type requires to provide CA certificate when a service is configured to use certificate
+signed by a CA different from the OpenShift CA to verify the service's certificate.
+And when the same CA is used by the administrator for signing several services'
+certificates it is unnecessary to configure the same CA certificate content as destinationCA
+in every created route object. And is also tiresome task having to rotate the
+certificate when expired, compromised or for any other reason in each route object. Instead,
+if an administrator could define a ConfigMap with the list of CA certificates
+and OpenShift Ingress Controller used it along with OpenShift CA to create a common 
+CA bundle for verifying service's certificates it would make managing routes a bit simpler.
+
+### User Stories
+- As an OpenShift cluster administrator, I want to configure admin or the enterprise
+  CA certificate bundle which should be used for verifying service's certificate by 
+  the OpenShift router, instead of configuring the same as destination CA certificate 
+  in each of the created route object.
+- As an OpenShift engineer, I want to create a CA trust bundle for Ingress operator usage,
+  which includes the OpenShift service CA and the admin provided CA bundle as the default
+  CA for verifying the service's certificate when a destination CA certificate is not
+  configured for a route.
+- As an OpenShift engineer, I want to update the OpenShift router certificate cache whenever
+  the OpenShift service CA certificate or the admin configured CA certificate bundle is 
+  updated or deleted.
+
+### Goals
+
+- An OpenShift administrator can configure a CA certificate bundle and expect
+  that to be used for verifying the service's certificate by the OpenShift
+  router without defining the destination CA certificate in the route object
+  created for the service.
+- Openshift Router should be updated seamlessly when the OpenShift service
+  CA certificate or the administrator configured CA certificate bundles are modified.
```

Comment:
I think we need a goal to have some way to make this new feature optional. In other words we have to support both the Route configured with its own, and the Route configured to use the new trusted bundle in the ConfigMap.  We also need to define which one will be used if both are present.

Reply:
Both the ways will be supported, if a Route is created with destinationCA then just that will be used for verifying server certificate, else the defaultDestinationCA will be used, which will contain both the user provided CA certificates and the OpenShift CA certificate.

Reply:
We can't have the default CA within the user-configured CA bundle.  Some routes should not be configured to have access to the user-configured CA if the route's `destinationCACertificate` is empty, that is not secure.  See https://github.com/openshift/enhancements/pull/1514/files#r1411110362